lib/src/rules/implicit_reopen.dart

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Don't implicitly reopen classes or mixins";
+
+/// todo(pq): link out to the spec or dart.dev docs.
+const _details = r'''
+Using the `interface`, `final`, `base`, `mixin`, and `sealed` class modifiers,
+authors can control whether classes and mixins allow being implemented,
+extended, and/or mixed in from outside of the library where they're defined.
+In some cases, it's possible for an author to inadvertantly relax these controls
+and implicitly "reopen" a class or mixin. This lint guards against that,
```

Comment:
> This lint guards against that, ...

Consider starting a new paragraph at this point, and expanding on "that", something like

This lint guards against unintentionally reopening a type by

Reply:
Great. Thanks!